hi-ml-histopathology/testhisto/testhisto/datamodules/test_datamodule_caching.py

```diff
@@ -177,3 +180,34 @@ def test_tile_id_coverage(mock_data_dir: Path, cache_mode: CacheMode, precache_l
                 f"Tile IDs already seen: {bag_tile_ids}"
             loaded_tile_ids.update(bag_tile_ids)
     assert loaded_tile_ids == expected_tile_ids
+
+
+def compare_bag_size(dl: DataLoader, bag_size: int) -> None:
+    for batch in dl:
+        for bag in batch:
+            assert len(bag) <= bag_size
+
+
+@pytest.mark.parametrize('cache_mode, precache_location, cache_dir_provided',
+                         [(CacheMode.DISK, CacheLocation.SAME, True),
+                          (CacheMode.DISK, CacheLocation.CPU, True),
+                          (CacheMode.MEMORY, CacheLocation.SAME, True),
+                          (CacheMode.MEMORY, CacheLocation.CPU, True),
+                          (CacheMode.MEMORY, CacheLocation.NONE, False),
+                          (CacheMode.NONE, CacheLocation.NONE, False)
+                          ])
```

Comment:
The cache consistency is tested above. For this specific test why we need to parametrize over all the options? Shouldn't this be independent of the cache mode?